src/palette/internal/palettecreator.cpp

```diff
@@ -1671,3 +1674,23 @@ PalettePtr PaletteCreator::newFretboardDiagramPalette()
 
     return sp;
 }
+
+PalettePtr PaletteCreator::newHarpPalette()
+{
+    PalettePtr sp = std::make_shared<Palette>(Palette::Type::Harp);
+    sp->setName(QT_TRANSLATE_NOOP("palette", "Harp notation"));
+    sp->setGridSize(90, 30);
+    sp->setDrawGrid(true);
+    sp->setVisible(true);
+
+    auto pedalDiagram = Factory::makeHarpPedalDiagram(gpaletteScore->dummy()->segment());
+    pedalDiagram->updateDiagramText();
+    sp->appendElement(pedalDiagram, QT_TRANSLATE_NOOP("palette", "Harp pedal diagram"));
+
+    auto pedalTextDiagram = Factory::makeHarpPedalDiagram(gpaletteScore->dummy()->segment());
+    pedalTextDiagram->setIsDiagram(false);
+    pedalTextDiagram->updateDiagramText();
```

Comment:
You could also consider to call `updateDiagramText()` in `layout()`, so that you don't need to remember to call it and so that it will always be up-to-date when other diagrams earlier in the score are modified.

(If you decide to do so, then my comment in src/engraving/libmscore/chordrest.cpp doesn't apply anymore)